modules/sdk-core/src/bitgo/staking/stakingWallet.ts

```diff
@@ -410,13 +408,62 @@ export class StakingWallet implements IStakingWallet {
           );
         }
       }
-      if (mismatchErrors.length > 0) {
-        const errorMessage = `Staking transaction validation failed before signing: ${mismatchErrors.join('; ')}`;
-        debug(errorMessage);
-        throw new Error(errorMessage);
+    }
+    if (buildParams?.memo && (explainedTransaction as any).memo !== buildParams.memo) {
+      mismatchErrors.push(
+        `Memo mismatch. Expected: '${JSON.stringify(buildParams.memo)}', Got: '${JSON.stringify(
+          (explainedTransaction as any).memo
+        )}'`
+      );
+    }
+
+    if (buildParams?.gasLimit && String((explainedTransaction as any).gasLimit) !== String(buildParams.gasLimit)) {
+      mismatchErrors.push(
+        `Gas Limit mismatch. Expected: ${buildParams.gasLimit}, Got: ${(explainedTransaction as any).gasLimit}`
+      );
+    }
+
+    if (buildParams?.type && (explainedTransaction as any).type !== buildParams.type) {
+      mismatchErrors.push(
+        `Transaction type mismatch. Expected: '${buildParams.type}', Got: '${(explainedTransaction as any).type}'`
+      );
+    }
+
+    if (buildParams?.solInstructions) {
+      if (
+        JSON.stringify((explainedTransaction as any).solInstructions) !== JSON.stringify(buildParams.solInstructions)
```

Comment:
I wonder if using JSON.stringify for deep comparison is actually reliable here? For example, { a: 1, b: 2} should equal { b: 2, a: 1} but the current comparison would produce otherwise.